src/java/src/main/java/com/dnanexus/DXSearch.java

```diff
@@ -989,14 +1022,110 @@ private T getDataObjectInstanceFromResult(FindDataObjectsResponse.Entry e) {
             return (T) dataObject;
         }
 
-
         @Override
         public Iterator<T> iterator() {
             return new ResultIterator();
         }
     }
 
     /**
+     * The page subset of data objects that matched a {@code findDataObjects} query.
+     *
+     * @param <T> data object class to be returned
+     */
+    public static class SearchPage<T extends DXDataObject> implements Iterable<T> {
```

Comment:
Thanks, this looks fine to me. The base class only seems useful if there is a way for clients to consume search results generically regardless of their type. Not sure if this will be useful, but we can always introduce it later by inserting it as a superclass.